nixos/modules/system/boot/systemd-unit-options.nix

```diff
@@ -263,12 +263,37 @@ in rec {
       description = "Shell commands executed as the service's main process.";
     };
 
+    scriptFullPrivileges = mkOption {
```

Comment:
I fear this would encourage contributors to write more imperative code, instead of the declarative way that should be preferred in modules. I acknowledge there are cases where a script requires full privileges to run, but suggest that the situations where this is the case are few enough that the cons of this change outweigh the pros.

This comment applies to all `*FullPrivileges` options introduced.

Reply:
My fear was that without it, it would lead to module local hacks instead - this is how I personally solved it in a yet-to-be-finished module that I worked on before this. Maybe I'm wrong, though. I would of course prefer if these options weren't needed at all.

Reply:
I'm always curious to see situations where something can't be achieved in a declarative way with NixOS because then we can see patterns emerge and come up with declarative solutions, like the author of `services.mysql.ensureDatabases` did.

Keep in mind that module local hacks can always be cleaned up, but once you add `*FullPrivileges` functions you're stuck with that interface for better or worse.

Reply:
That's true - I guess one way forward would be to convert this into a local hack and look at cleaning it up later, since I would prefer if this could be done in time for 19.09...

Reply:
If we buckle down on this I think we could get it done in time for `19.09` in a declarative way, without the need for escalated scripts. I'm happy to chip in.